spring-integration-zookeeper/src/main/java/org/springframework/integration/zookeeper/lock/ZookeeperLockRegistry.java

```diff
@@ -300,10 +299,16 @@ public Boolean call() throws Exception {
 					return this.mutex.acquire(waitTime, TimeUnit.MILLISECONDS);
 				}
 			}
-			catch (TimeoutException e) {
-				future.cancel(true);
+			catch (@SuppressWarnings("unused") TimeoutException e) {
+				if (future != null) {
+					future.cancel(true);
+				}
 				return false;
 			}
+			catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+				throw e;
+			}
```

Comment:
These two are bugs (need backporting).

Reply:
Can you push these fixes by yourself to those branches?
Or do you want me to extract them somehow from such a big change? 😄 

Thanks

Reply:
No; I'll do it after review; I just didn't want to forget.